can/io/asc.py

```diff
@@ -254,15 +254,17 @@ def _process_fd_can_frame(self, line: str, msg_kwargs: Dict[str, Any]) -> Messag
 
     def __iter__(self) -> Generator[Message, None, None]:
         self._extract_header()
-
+        trigger_match_re = re.compile(
```

Comment:
You can move these outside of the class as module constants, so they are only compiled once.

```python3
ASC_TRIGGER_REGEX: Final = re.compile(
    ...
)
ASC_MESSAGE_REGEX: Final = re.compile(
    ...
)
```